### PR TITLE
switch from boost::any to std::any

### PR DESCRIPTION
**Pull request by greglandrum**
Given that `std::any` exists in c++ 17, there's no reason to keep using boost::any

This also bumps the linux CI builds to use g++-10 due to a bug in v9.4 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90415)

### CODE REVIEW

File: Code/DataStructs/DatastructsStreamOps.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ class DataStructsExplicitBitVecPropHandler : public CustomPropHandler {
           rdvalue_cast<const ExplicitBitVect &>(value).toString();
       streamWrite(ss, output);
       return true;
-    } catch (boost::bad_any_cast &) {
+    } catch (std::bad_any_cast &) {
```
Comment (ricrogz) [inline Code/DataStructs/DatastructsStreamOps.h:61]: There's another `#include <boost/any.hpp>` at the top of this file (github doesn't allow me to highlight it too).

File: Code/GraphMol/FileParsers/SDWriter.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ void _writePropToStream(std::ostream *dp_ostream, const ROMol &mol,
   std::string pval;
   try {
     mol.getProp(name, pval);
-  } catch (boost::bad_any_cast &) {
+  } catch (std::bad_any_cast &) {
```
Comment (ricrogz) [inline Code/GraphMol/FileParsers/SDWriter.cpp:88]: There's also a `#include <boost/any.hpp>` include at the top of this file.

**Review by ricrogz — APPROVED** [review]
Should we remove `Code/Demos/boost/any_container/dict.cpp` too ?

Besides that, LGTM once the two remaining `#include <boost/any.hpp>` mentioned below.

**Comment by ricrogz** [comment]
> Besides that, LGTM once the two remaining `#include <boost/any.hpp>` mentioned below.

And the build errors, too :-P

**Comment by greglandrum** [comment]
> > Besides that, LGTM once the two remaining `#include <boost/any.hpp>` mentioned below.
> 
> And the build errors, too :-P

Yeah... those...
Unfortunately it looks like the current failures on ubuntu are due to a compiler bug in g++ 9.4 (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90415). I'm going to have to either switch the linux CI builds to use 10.5 (seemingly available in the ubuntu 20.04 VMs) or update them to use the 22.04 VMs.

Will do a bit of testing around this today.

**Comment by greglandrum** [comment]
@ricrogz : I made the changes you suggested and got the CI builds working (the most recent mac failure is just one of those intermittent timeouts that happen)

**Review by ricrogz — APPROVED** [review]
LGTM

**Comment by ricrogz** [comment]
> (the most recent mac failure is just one of those intermittent timeouts that happen)

Since these seem to be frequent now, should we try bumping the Mac build timeout to 2h ?

**Comment by greglandrum** [comment]
> > (the most recent mac failure is just one of those intermittent timeouts that happen)
> 
> Since these seem to be frequent now, should we try bumping the Mac build timeout to 2h ?

worth a try

**Pull request merged**